lib/fine-types/app/Options.hs

```diff
@@ -0,0 +1,76 @@
+{-# LANGUAGE ApplicativeDo #-}
+{-# LANGUAGE LambdaCase #-}
+
+module Options where
+
+import Prelude
+
+import Options.Applicative
+    ( ReadM
+    , command
+    , eitherReader
+    , execParser
+    , fullDesc
+    , header
+    , help
+    , helper
+    , hsubparser
+    , info
+    , long
+    , metavar
+    , option
+    , progDesc
+    , short
+    , strArgument
+    , (<**>)
+    )
+
+newtype Options = Options
+    { optCommand :: Commands
+    }
+
+newtype Commands
+    = Convert ConvertOptions
+
+data Format = Json | Yaml
+
+data ConvertOptions = ConvertOptions
+    { optInput :: FilePath
+    , optOutput :: FilePath
+    , optFormat :: Format
+    }
+
+parseOptions :: IO Options
+parseOptions =
+    execParser
+        $ info (options <**> helper)
+        $ fullDesc
+            <> progDesc "Run fine-types commands"
+            <> header "fine-types - a tool for working with fine-types modules"
+  where
+    options = Options <$> commands
+
+    commands =
+        hsubparser
+            $ mconcat
+                [ command "convert" $ info (Convert <$> convertOptions) mempty
+                -- other commands go here
+                ]
+
+    convertOptions = do
+        input <- strArgument $ metavar "FILE" <> help "input file"
+        output <- strArgument $ metavar "FILE" <> help "output file"
+        format <-
+            option formatReader
+                $ long "format"
+                    <> short 'f'
```

Comment:
```suggestion
```

I have tried to find common conventions for command line arguments, but they are a bit sparse:

* [Linux Documentation Project: Standard Command-Line Options](https://tldp.org/LDP/abs/html/standard-options.html)
* [Eric S. Raymond: Command-Line Options](http://catb.org/~esr/writings/taoup/html/ch10s05.html)

In these documents, the short form `-f` is mentioned, but for a different effect like `file` or `force`. The long option is fine, but I would like to disallow the short form here.